plugins/destination/duckdb/client/write.go

```diff
@@ -202,59 +183,77 @@ func transformRecord(sc *arrow.Schema, rec arrow.Record) arrow.Record {
 }
 
 func transformArray(arr arrow.Array) arrow.Array {
-	dt := arr.DataType()
-	switch {
-	case arrow.TypeEqual(dt, types.ExtensionTypes.UUID) ||
-		arrow.TypeEqual(dt, types.ExtensionTypes.Inet) ||
-		arrow.TypeEqual(dt, types.ExtensionTypes.MAC) ||
-		arrow.TypeEqual(dt, types.ExtensionTypes.JSON) ||
-		dt.ID() == arrow.STRUCT:
+	if plainWrite(arr.DataType()) {
+		return arr
+	}
+
+	switch arr := arr.(type) {
+	case *types.UUIDArray, *types.InetArray, *types.MACArray, *types.JSONArray, *array.Struct:
 		return transformToStringArray(arr)
-	case arrow.TypeEqual(dt, arrow.PrimitiveTypes.Uint8):
-		return transformUint8ToUint32Array(arr.(*array.Uint8))
-	case arrow.TypeEqual(dt, arrow.PrimitiveTypes.Uint16):
-		return transformUint16ToUint32Array(arr.(*array.Uint16))
-	case arrow.IsListLike(dt.ID()):
-		child := transformArray(arr.(*array.List).ListValues()).Data()
+	case *array.Uint8:
+		return transformUint8ToUint32Array(arr)
+	case *array.Uint16:
+		return transformUint16ToUint32Array(arr)
+	case *array.Timestamp:
+		// mismatching unit or tz
+		return transformTimestamp(duckDBToArrow(arrowToDuckDB(arr.DataType())).(*arrow.TimestampType), arr)
```

Comment:
here we know that duckdb supports timestamps, but the `plainWrite` is false => the unit or time zone doesn't match. So, we transform the values to the proper unit/zone